File: exe/lp/lista5.py
## Lista 5 de LP - Parte 1
## Douglas Rodrigues de Almeida


from math import pi
from sys import exit

class Forma:
        def area(self):
                print("area nao implementada")

        def perimetro(self):
                perint("perimetro nao iplementado")

class Quadrado(Forma):
        def __init__(self):
                self.lado = float(input("Digite valor (para lado do quadrado): "))

        def area(self):
                return self.lado * self.lado

        def nome(self):
                return "quadrado"

        def perimetro(self):
                return self.lado * 4

class Circulo(Forma):
        def __init__(self):
                self.raio = float(input("Digite valor (para raio do circulo): "))
        
        def area(self):
                return pi * self.raio * self.raio
        
        def nome(self):
                return "circulo"

        def perimetro(self):
                return 2 * pi * self.raio

class Triangulo(Forma):
        def __init__(self):
                self.lado1 = float(input("Digite valor (para lado 1 do triangulo): "))
                self.lado2 = float(input("Digite valor (para lado 2 do triangulo): "))
                self.lado3 = float(input("Digite valor (para lado 3 do triangulo): "))

        def area(self):
                s = self.perimetro() / 2.0
                return s * (s - self.lado1) * (s - self.lado2) * (s - self.lado3)

        def nome(self):
                return "triangulo"

        def perimetro(self):
                return self.lado1 + self.lado2 + self.lado3

def main():
        f = input("Digite q para quadrado, c para circulo ou t para triangulo: ")
        while True:
                if f == 'q':
                        forma = Quadrado()
                        break
                elif f == 'c':
                        forma = Circulo()
                        break
                elif f == 't':
                        forma = Triangulo()
                        break
                else:
                        sys.exit()
        
        print("\tarea do {0} igual a {1}".format(forma.nome(), forma.area()))
        print("\tperimetro do {0} igual a {1}".format(forma.nome(), forma.perimetro()))

main()

File: exe/lp/lista5.hs
-- Lista 5 de LP - Parte 3
-- Douglas Rodrigues de Almeida


type Lado = Float
type Raio = Float
data Forma = Quadrado Lado | Circulo Raio | Triangulo Lado Lado Lado

area (Quadrado lado) = lado * lado
area (Circulo raio) = pi * raio * raio
area (Triangulo lado1 lado2 lado3) =
                                let s = (lado1 + lado2 + lado3) / 2
                                in s * (s - lado1) * (s - lado2) * (s - lado3)
perimetro (Quadrado lado) = 4 * lado
perimetro (Circulo raio) = 2 * pi * raio
perimetro (Triangulo lado1 lado2 lado3) = lado1 + lado2 + lado3

quadrado:: IO Forma
quadrado = do
        putStrLn "Digite valor (para lado do quadrado): "
        s <- getLine
        return . Quadrado $ read s

circulo:: IO Forma
circulo = do
        putStrLn "Digite valor (para raio do circulo): "
        s <- getLine
        return . Circulo $ read s

triangulo:: IO Forma
triangulo = do
        putStrLn "Digite valor (para lado 1 do triangulo): "
        x <- getLine
        putStrLn "Digite valor (para lado 2 do triangulo): "
        y <- getLine
        putStrLn "Digite valor (para lado 3 do triangulo): "
        z <- getLine
        return (Triangulo (read x) (read y) (read z))

main = do
        putStrLn "Digite q para quadrado, c para circulo ou t para triangulo: "
        f <- getChar
        getLine
        let (nome, criarForma)
                | f == 'q' = ("quadrado", quadrado)
                | f == 'c' = ("circulo", circulo)
                | f == 't' = ("triangulo", triangulo)
        forma <- criarForma
        let areaForma = show $ area forma
        putStrLn ("area do " ++ nome ++ " igual a " ++ areaForma)
        let perimetroForma = show $ perimetro forma
        putStrLn ("perimetro do " ++ nome ++ " igual a " ++ perimetroForma)